### PR TITLE
Updating SDL keyboard to use new KOS API.

### PR DESCRIPTION
**Pull request by gyrovorbis**
This needs to be merged as soon as https://github.com/KallistiOS/KallistiOS/pull/503 goes live to make it work with the SDL kos-port.

### CODE REVIEW

File: SDL/files/SDL_dcevents.c [hunk 1]
```diff
@@ -101,7 +101,7 @@ static void mouse_update(void) {
 }
 
 static void keyboard_update(void) {
-    static kbd_state_t old_state;
+	static kbd_mods_t old_mods;
```
Comment (bucanero) [inline SDL/files/SDL_dcevents.c:104]:
```suggestion
    static kbd_mods_t old_mods;
```

It seems they're using spaces 😉

**Comment by QuzarDC** [comment]
@gyrovorbis Wanna use this as an opportunity to flex the new KOS version testing functionality? That way the update won't break people coming in on v2.1.0.

Though, now that makes me realize we'd want to do stuff like try to test ports changes against versions, but that should rarely be much of an issue.

**Comment by gyrovorbis** [comment]
> @gyrovorbis Wanna use this as an opportunity to flex the new KOS version testing functionality? That way the update won't break people coming in on v2.1.0.
> 
> Though, now that makes me realize we'd want to do stuff like try to test ports changes against versions, but that should rarely be much of an issue.

Not a bad idea! Done!

**Review by QuzarDC — APPROVED** [review]

**Comment by ppxxcc** [comment]
I really don't want to be snarky but would it be too much trouble for the authors of PRs to spend the 10 seconds to rebuild and make sure that the **source code which has been modified** actually builds? Presumably you are already sitting there at your computer with the modified source code ready because how else would you have just pushed a change to your local repo which you are about to submit?

This isn't the first time went blatantly untested changes (which don't even build, should be so obvious!) broke KOS or kos-ports for everybody else.

**Comment by gyrovorbis** [comment]
The issues with the versioning system (which were reviewed by half a dozen developers and wasn't caught by anyone, so sue us) was totally fixed here: https://github.com/KallistiOS/kos-ports/pull/70, which I was expecting to get merged first. 

Almost like CI would ensure this would never happen again or something.

**Comment by QuzarDC** [comment]
> The issues with the versioning system (which were reviewed by half a dozen developers and wasn't caught by anyone, so sue us) was totally fixed here: #70, which I was expecting to get merged first.
> 
> Almost like CI would ensure this would never happen again or something.
> 
> My mistake was not communicating that the other PR needed to be merged first due to this issue, not "zomg bro never even built this thing."

#70 is this PR. Which were you meaning needed to be merged first?

**Comment by darcagn** [comment]
> > The issues with the versioning system (which were reviewed by half a dozen developers and wasn't caught by anyone, so sue us) was totally fixed here: #70, which I was expecting to get merged first.
> > Almost like CI would ensure this would never happen again or something.
> > My mistake was not communicating that the other PR needed to be merged first due to this issue, not "zomg bro never even built this thing."
> 
> #70 is this PR. Which were you meaning needed to be merged first?

I believe he means https://github.com/KallistiOS/KallistiOS/pull/503 which has changes to `version.h`. But this being merged would not solve the situation, because `version.h` would still be broken in KOS stable, breaking SDL compilation. Before the SDL kos-port can be updated I believe the `version.h` fix would need to be backported to stable and a new v2.1.1 tag released.

**Pull request merged**